prov/verbs/src/fi_verbs.c

```diff
@@ -389,89 +383,244 @@ ssize_t fi_ibv_send_iov_flags(struct fi_ibv_msg_ep *ep, struct ibv_send_wr *wr,
+				 (size_t *)&fi_ibv_gl_data.rdm.buffer_size) ||
+	    (fi_ibv_gl_data.rdm.buffer_size < sizeof(struct fi_ibv_rdm_rndv_header))) {
+		VERBS_INFO(FI_LOG_CORE,
+			   "rdm_buffer_size should be greater then %d\n",
+			   sizeof(struct fi_ibv_rdm_rndv_header));
+		return -FI_EINVAL;
+	}
+	if (fi_ibv_get_param_int("rdm_rndv_seg_size", "The segment size for "
+				 "zero copy protocols (bytes)",
+				 (size_t *)&fi_ibv_gl_data.rdm.rndv_seg_size) ||
+	    (fi_ibv_gl_data.rdm.rndv_seg_size <= 0)) {
+		VERBS_INFO(FI_LOG_CORE,
+			   "Invalid value of rdm_rndv_seg_size\n");
+		return -FI_EINVAL;
+	}
+	if (fi_ibv_get_param_int("rdm_thread_timeout", "The wake up timeout of "
+				 "the helper thread (usec)",
+				 (size_t *)&fi_ibv_gl_data.rdm.thread_timeout) ||
+	    (fi_ibv_gl_data.rdm.thread_timeout < 0)) {
+		VERBS_INFO(FI_LOG_CORE,
+			   "Invalid value of rdm_thread_timeout\n");
+		return -FI_EINVAL;
+	}
+	if (fi_ibv_get_param_str("rdm_eager_send_opcode", "The operation code that "
+				 "will be used for eager messaging. Only IBV_WR_SEND "
+				 "and IBV_WR_RDMA_WRITE_WITH_IMM are supported. "
+				 "The last one is not applicable for iWarp.",
+				 &fi_ibv_gl_data.rdm.eager_send_opcode))
+		return -FI_EINVAL;
+
+	/* DGRAM-specific parameters */
+	if (getenv("OMPI_COMM_WORLD_RANK") || getenv("PMI_RANK"))
+		fi_ibv_gl_data.dgram.use_name_server = 0;
+	if (fi_ibv_get_param_bool("dgram_use_name_server", "The option that "
+				  "enables/disables OFI Name Server thread that is used "
+				  "to resolve IP-addresses to provider specific "
+				  "addresses. If MPI is used, the NS is disenabled "
+				  "by default.", &fi_ibv_gl_data.dgram.use_name_server))
+		return -FI_EINVAL;
+	if (fi_ibv_get_param_int("dgram_name_server_port", "The port on which Name Server "
+				 "thread listens incoming connection and requestes.",
+				 (size_t *)&fi_ibv_gl_data.dgram.name_server_port) ||
+	    (fi_ibv_gl_data.dgram.name_server_port < 0 ||
+	     fi_ibv_gl_data.dgram.name_server_port > 65535)) {
+		VERBS_INFO(FI_LOG_CORE,
+			   "Invalid value of dgram_name_server_port\n");
+		return -FI_EINVAL;
+	}
+	if (fi_ibv_get_param_bool("dgram_separate_send_recv_cq", "Creates two separate "
+				  "Send/Recv CQ.",
+				  &fi_ibv_gl_data.dgram.separate_send_recv_cq))
+		return -FI_EINVAL;
+	if (fi_ibv_get_param_int("dgram_device_port_number", "Port number of device to be "
+				 "only used for generating fi_info. The parameter is "
+				 "ignored if device name wasn't specified.",
+				 (size_t *)&fi_ibv_gl_data.dgram.device.port_number) ||
+	    (fi_ibv_gl_data.dgram.device.port_number < 0 ||
+	     fi_ibv_gl_data.dgram.device.port_number > 32)) {
+		VERBS_INFO(FI_LOG_CORE,
+			   "Invalid value of dgram_device_port_number\n");
+		return -FI_EINVAL;
+	}
+	if (fi_ibv_get_param_str("dgram_device_name", "The name of device to be "
+				 "only used for generating fi_info.",
+				 &fi_ibv_gl_data.dgram.device.name))
+		return -FI_EINVAL;
+
+	return FI_SUCCESS;
```

Comment:
I don't think we want failures setting or getting environment variables to disable the provider.  I prefer that we simply log an error message and continue using the default settings.

Reply:
Ok, I agree. But we can face with the cases when some ordinary user that uses verbs, set an incorrect value to some param and doesn't have thoughts to check log output, will think that he/she works with proper setting, but it isn't so